oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/MetadataQueryEsDAO.java

```diff
@@ -61,80 +60,66 @@ public MetadataQueryEsDAO(ElasticSearchClient client, int queryMaxSize) {
 
-        final BoolQueryBuilder query = Query.bool().must(
-            Query.term(ServiceTraffic.NODE_TYPE, NodeType.Database.value()));
-        final SearchBuilder search = Search.builder().query(query).size(queryMaxSize);
-        final SearchResponse results = getClient().search(index, search.build());
+            Endpoint endpoint = new Endpoint();
+            endpoint.setId(endpointTraffic.id());
+            endpoint.setName((String) sourceAsMap.get(EndpointTraffic.NAME));
+            endpoints.add(endpoint);
+        }
 
-        final List<Service> serviceList = buildServices(results);
-        return serviceList.stream().map(service -> {
-            Database database = new Database();
-            database.setId(service.getId());
-            database.setName(service.getName());
-            return database;
-        }).collect(Collectors.toList());
+        return endpoints;
     }
 
     @Override
-    public List<Service> searchServices(final NodeType nodeType, final String keyword) throws IOException {
+    public List<Service> listServices(final String layer, final String group) throws IOException {
         final String index =
             IndexController.LogicIndicesRegister.getPhysicalTableName(ServiceTraffic.INDEX_NAME);
 
         final BoolQueryBuilder query =
-            Query.bool()
-                 .must(Query.term(ServiceTraffic.NODE_TYPE, nodeType.value()));
+            Query.bool();
         final SearchBuilder search = Search.builder().query(query).size(queryMaxSize);
-
-        if (!Strings.isNullOrEmpty(keyword)) {
-            String matchCName = MatchCNameBuilder.INSTANCE.build(ServiceTraffic.NAME);
-            query.must(Query.match(matchCName, keyword));
+        if (StringUtil.isNotEmpty(layer)) {
+            query.must(Query.term(ServiceTraffic.LAYER, Layer.valueOf(layer).value()));
         }
+        if (StringUtil.isNotEmpty(group)) {
+            query.must(Query.term(ServiceTraffic.GROUP, group));
+        }
+        final SearchResponse results = getClient().search(index, search.build());
 
-        SearchResponse response = getClient().search(index, search.build());
-        return buildServices(response);
+        return buildServices(results);
     }
 
     @Override
-    public Service searchService(final NodeType nodeType, final String serviceCode) throws IOException {
+    public Service findService(final String serviceId) throws IOException {
         final String index =
             IndexController.LogicIndicesRegister.getPhysicalTableName(ServiceTraffic.INDEX_NAME);
         final BoolQueryBuilder query =
             Query.bool()
-                 .must(Query.term(ServiceTraffic.NODE_TYPE, nodeType.value()))
-                 .must(Query.term(ServiceTraffic.NAME, serviceCode));
+                 .must(Query.term(ServiceTraffic.SERVICE_ID, serviceId));
         final SearchBuilder search = Search.builder().query(query).size(1);
```

Comment:
I am feeling `size(1)` is not correct here anymore. We could have multiple layers(meaning multiple records per service) now.